src/settings.rs

```
@@ -208,20 +214,30 @@ impl ContextSettingInternalExt for Context {
 }
 
 impl ContextSettingExt for Context {
-    fn get_setting_properties<T: SettingType + 'static>(&self) -> SettingProperties {
-        let data_container = self
-            .get_data_container(SettingDataPlugin)
-            .unwrap()
+    fn get_setting_properties<T: SettingType + 'static>(
+        &self,
+    ) -> Result<SettingProperties, IxaError> {
+        let data_container = self.get_data_container(SettingDataPlugin).unwrap();
```

It seems like we should raise an error if the setting data plugin is None as well -- because that would mean no settings had been registered. So, instead of calling `unwrap` here, I would call `ok_or(Err(IxaError::IxaError("your error message").to_string()))`.

Honestly, I think trying to get setting properties for non registered settings is deserving of a panic rather than an error, but I like having the validation checks so I'm good with it for now. Also, settings will look differently later on, and I think this validation will hopefully become not necessary once we wrap some of the setting data containers in RefCells and do lazy initialization.